src/server/jobs/update-bcast.js

```diff
@@ -262,11 +265,27 @@ async function main() {
 
           // For each listing, push new database record into array
           for (const mediaEntry of mediaEntries) {
+            const mediaEntryDate = mediaEntry['date for recording(s)'];
+            let weekNum = weekNumCache.get(mediaEntryDate);
```

Comment:
@chern This isn't the caching I mentioned. You want to cache the expensive parts of the code. So the processing logic of figuring out if a date follows within a given time period is not the expensive part. The API call to the Registrar web service is the expensive part. Especially when the term data doesn't change in between calls.

Caching at this part will hardly save any API calls, because how often do medias share the same date?

Cache the API call in getWeekNumber() and the resulting data calculations so it is only processed once.